libs/sdk-ui-filters/src/AttributeFilter@next/hooks/useAttributeFilterController.ts

```diff
@@ -138,8 +141,19 @@ function useInitOrReload(
     }, [handler]);
 
     useEffect(() => {
+        // filter object has been changed
         if (!isEqual(filter, handler.getFilter())) {
-            handler.init();
+            //check if filter refs are different
+            if (isEqual(filterObjRef(filter), filterObjRef(handler.getFilter()))) {
+                //if its refs are equal need to set selection init is not necessary
+                const elements = filterAttributeElements(filter);
+                const keys = isAttributeElementsByValue(elements) ? elements.values : elements.uris;
+                const isInverted = isNegativeAttributeFilter(filter);
+                handler.changeSelection({ keys, isInverted });
```

Comment:
This can be problematic (not in AD/KD). Imagine, that someone changes the filter to the values, that are not loaded inside the AttributeFilterHandler. This is why init is here -> inconsistent internal state and what is coming to it.

Reply:
fixed